driver/src/main/java/org/neo4j/driver/internal/NetworkSession.java

```diff
@@ -155,28 +164,41 @@ public void close()
                 }
             }
         }
-        
+
         syncAndCloseCurrentConnection();
     }
 
     @Override
-    public Transaction beginTransaction()
+    public synchronized Transaction beginTransaction()
     {
-        return beginTransaction( null );
+        return beginTransaction( mode );
     }
 
     @Override
     public synchronized Transaction beginTransaction( String bookmark )
     {
-        ensureSessionIsOpen();
-        ensureNoOpenTransactionBeforeOpeningTransaction();
+        setLastBookmark( bookmark );
```

Comment:
Good point, I just tried to use same `#setLastBookmark()` method everywhere and it prohibits nulls. Will fix.